pressbooks/pressbooks-network-catalog
# fix: vite 4 build, apostrophe handling

This PR fix #203 , #181 and #200

I removed the unneeded filter_var for the request parameters those are already sanitized by Laravel's default sanitization methods and I added a couple of `stripslashes` to avoid having issues with items with apostrophes in the UI, the string will be sanitized in the wp:prepare method

I modified how the manifest entry point was mapped due some rollup changes in terms of css processing in Vite 4.

I had to rename the `search` get parameter used in the catalog it seems is causing some issues with WP, because is probably is being used in other wp hooks to paginate blog posts or pages when is running as frontpage.

How to test:

1. Test the build process works on dev and production mode. (modifying the value of) `IS_VITE_DEVELOPMENT` in `pressbooks-network-catalog.php`
2. Test adding an institution with apostrophes or any special character like `St. Angela's College` you should be able to filter by that institutions and the filter it should be applied and selected on the sidebar.
3. This PR also address the duplicated parameters on desktop (those are only the duplicated selects that we are using for mobile) I mean those are not harmful just annoying to see on desktop maybe
4. Apply a term search when the catalog is set as homepage the catalog should work as expected
